include/git2/remote.h

```diff
@@ -474,6 +474,17 @@ typedef int GIT_CALLBACK(git_push_negotiation)(const git_push_update **updates,
  */
 typedef int GIT_CALLBACK(git_push_update_reference_cb)(const char *refname, const char *status, void *data);
 
+/**
+ * Callback to resolve URLs before connecting to remote
+ *
+ * @param url The URL to resolve
+ * @param url_resolved The resolved URL
+ * @param direction GIT_DIRECTION_FETCH or GIT_DIRECTION_PUSH
+ * @param payload The callback payload
+ * @return 0 on success, otherwise an error
+ */
+typedef int GIT_CALLBACK(git_resolve_url_cb)(const char *url, char **url_resolved, int direction, void *payload);
```

Comment:
The output-parameter `url_resolved` should be the first argument to this function